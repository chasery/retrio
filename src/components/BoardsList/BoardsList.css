.BoardsList {
  display: grid;
  grid-template-columns: 1fr;
  column-gap: 1rem;
  row-gap: 1rem;
  margin: 1rem 0 0;
  padding: 0;
}
.BoardsList > li {
  align-self: start;
  list-style: none;
  box-shadow: 0 0.0625rem 0.0625rem #0000001c;
  background-color: #fff;
}

@media all and (min-width: 30rem) {
  .BoardsList {
    grid-template-columns: repeat(2, calc(50% - 0.5rem));
  }
}

@media all and (min-width: 50rem) {
  .BoardsList {
    grid-template-columns: repeat(3, calc(33.333% - 0.667rem));
  }
}
