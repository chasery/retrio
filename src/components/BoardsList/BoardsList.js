import React from 'react';
import BoardsItem from '../BoardsItem/BoardsItem';
import './BoardsList.css';

function BoardsList(props) {
  const { boards, deleteBoard } = props;

  const renderBoards = (boards) => {
    if (boards) {
      return boards.map((board) => (
        <BoardsItem
          key={board.id}
          id={board.id}
          name={board.name}
          owner={board.owner}
          teamId={board.team_id}
          createdAt={board.created_at}
          updatedAt={board.updated_at}
          deleteBoard={deleteBoard}
        />
      ));
    }
  };

  return <ol className='BoardsList'>{renderBoards(boards)}</ol>;
}

export default BoardsList;
