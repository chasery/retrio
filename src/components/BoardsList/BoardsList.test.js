import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import BoardsList from './BoardsList';

describe('BoardsList component', () => {
  it('renders without error', () => {
    const wrapper = shallow(<BoardsList />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
