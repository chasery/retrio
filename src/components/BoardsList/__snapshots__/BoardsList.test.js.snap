// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BoardsList component renders without error 1`] = `
<ol
  className="BoardsList"
/>
`;
